docker/docker_image_src.go

```diff
@@ -39,6 +39,19 @@ func newImageSource(ctx *types.SystemContext, ref dockerReference, requestedMani
 	if requestedManifestMIMETypes == nil {
 		requestedManifestMIMETypes = manifest.DefaultRequestedManifestMIMETypes
 	}
+	var (
+		mts   = supportedManifestMIMETypesMap()
+		reset = true
+	)
+	for _, mtrequested := range requestedManifestMIMETypes {
+		if !mts[mtrequested] {
+			continue
+		}
+		reset = false
```

Comment:
`break` here, and then it’s more natural to reword as `if mts[mtrequested] { `**EDIT reset = false;**` break } `.